src/main/scala/rocketchip/PassManager.scala

```diff
@@ -0,0 +1,61 @@
+// See LICENSE for license details.
+
+package rocketchip
+
+import firrtl._
+import firrtl.ir._
+import firrtl.passes.Pass
+import firrtl.Annotations.AnnotationMap
+import java.io.Writer
+
+// The Rocket Chip pass manager: a simple pass manager that allows various
+// FIRRTL passes to be inserted at various points throughout the FIRRTL build
+// process on the way to Verilog.  It's expected that users will extend this
+// class in order to specialize their passes.
+class RocketChipPassManager extends firrtl.Compiler {
+  private def convertToTransforms(ors: Seq[Either[Pass, Transform]]): Seq[Transform] = {
+    class SinglePassTransform(pass: Pass) extends Transform with SimpleRun {
+      def execute(circuit: Circuit, annotationMap: AnnotationMap): TransformResult =
+        run(circuit, Seq(pass))
+    }
+
+    def convertToTransform(either: Either[Pass, Transform]): Transform =
+      either match {
+        case Right(t) => t
+        case Left(p)  => new SinglePassTransform(p)
+      }
+
+    ors.map( or => convertToTransform(or) )
+  }
+
+  // This is some sugar to users don't need to actually know ther's a tagged
+  // union in here to allow them to transparently provide either Pass or
+  // Transform instances to the pass manager.
+  implicit def l2Either[L, R](l: L): Either[L, R] = Left(l)
+  implicit def r2Either[L, R](r: R): Either[L, R] = Right(r)
+
+  // These functions are designed to be overridden by users.  They return a
+  // sequence of transformations that will be applied at various FIRRTL levels.
+  // A transform that operates at a FIRRTL level is expected to both produce
+  // and consume FIRRTL at that level.
```

Comment:
I think this makes sense but I also I think people want to write passes that consume low/mid/high and produce high, for convenience, but I'm not sure if the HighToMid And MidToLow are idempotent. I'm sure I read somewhere that they are expected to be eventually but I don't know the current status.